docs/apidocs/synthesis_discretebasis.rst

```diff
@@ -0,0 +1,6 @@
+.. _qiskit-synthesis-discrete_basis:
+
+.. automodule:: qiskit.synthesis.discrete_basis
+   :no-members:
+   :no-inherited-members:
+   :no-special-members:
```

Comment:
If you want this file built into the API documentation list, you'd probably want to add it to `docs/apidocs/index.rst`.  That said, it may make things more structured to instead add it into the docstring of `qiskit.synthesis` with an `.. automodule:: qiskit.synthesis.discrete_basis` line.